### ISSUE TITLE
[BUG] gpg.encrypt text=foo output=/some/file does not write to a file

### ISSUE THREAD

**Issue by oeuftete** [saltstack/salt#56645]
**Description**

**Setup**
(Please provide relevant configs and/or SLS files (be sure to remove sensitive info).

**Steps to Reproduce the behavior**

(Include debug logs if possible and relevant)

**Expected behavior**

**Screenshots**

**Versions Report**
<details><summary>salt --versions-report</summary>
(Provided by running salt --versions-report. Please also mention any differences in master/minion versions.) 

```
PASTE HERE
```
</details>

**Additional context**

**Comment by oeuftete** [comment]
Pretty straightforward to see.  The `text` case doesn't have any logic to output to file, the `filename` case does:

https://github.com/saltstack/salt/blob/5c450f3bce57ed9798c7c5ff06ffa391343c5b82/salt/modules/gpg.py#L1190-L1205

But either way, the return says the file was written if the result was ok.

https://github.com/saltstack/salt/blob/5c450f3bce57ed9798c7c5ff06ffa391343c5b82/salt/modules/gpg.py#L1209-L1212

**Comment by ritwikbadola** [comment]
Can I try and solve this issue ?

**Comment by oeuftete** [comment]
> Can I try and solve this issue ?

@ritwikbadola  Definitely!  Here's a link to information on contributing: https://docs.saltstack.com/en/latest/topics/development/contributing.html

**Comment by julianpacheco** [comment]
Hope you don't mind @ritwikbadola I just entered a PR to resolve this issue since it's still open.

**Comment by dwoz** [comment]
Fixed in 3006.0 by 59608a1d21f ("reworked encrypt function"). `gpg.encrypt(text=..., output=...)` now passes `output` through to python-gnupg; file is written and round-trips cleanly. Verified on 3006.x / 3007.x / 3008.x / master.
